### ISSUE TITLE
Zooming makes the screen smaller & smaller

### ISSUE THREAD

**Issue by Uvais-Mohammad** [yakagami/zoom_view#2]
```
import 'package:flutter/material.dart';
import 'package:zoom_view/zoom_view.dart';

import '../settings/settings_view.dart';
import 'sample_item.dart';
import 'sample_item_details_view.dart';

class SampleItemListView extends StatefulWidget {
  const SampleItemListView({
    super.key,
    this.items = const [
      SampleItem(1),
      SampleItem(2),
      SampleItem(3),
      SampleItem(4),
      SampleItem(1),
      SampleItem(2),
      SampleItem(3),
      SampleItem(4),
      SampleItem(1),
      SampleItem(2),
      SampleItem(3),
      SampleItem(4),
      SampleItem(1),
      SampleItem(2),
      SampleItem(3),
      SampleItem(4),
      SampleItem(1),
      SampleItem(2),
      SampleItem(3),
      SampleItem(4),
    ],
  });

  static const routeName = '/';

  final List<SampleItem> items;

  @override
  State<SampleItemListView> createState() => _SampleItemListViewState();
}

class _SampleItemListViewState extends State<SampleItemListView> {
  final controller = ScrollController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Sample Items'),
        actions: [
          IconButton(
            icon: const Icon(Icons.settings),
            onPressed: () {
              Navigator.restorablePushNamed(context, SettingsView.routeName);
            },
          ),
        ],
      ),
      body: ColoredBox(
        color: Colors.white,
        child: ZoomListView(
          child: ListView.builder(
            controller: controller,
            restorationId: 'sampleItemListView',
            itemCount: widget.items.length,
            itemBuilder: (BuildContext context, int index) {
              final item = widget.items[index];

              return Center(
                child: ListTile(
                    title: Text('SampleItem ${item.id}',
                        style: const TextStyle(color: Colors.black)),
                    leading: const CircleAvatar(
                      foregroundImage:
                          AssetImage('assets/images/flutter_logo.png'),
                    ),
                    onTap: () {
                      Navigator.restorablePushNamed(
                        context,
                        SampleItemDetailsView.routeName,
                      );
                    }),
              );
            },
          ),
        ),
      ),
    );
  }
}

```

https://github.com/yakagami/zoom_view/assets/63300484/8a43fad9-54c5-4d2f-9a34-bbe6b50bff7f

**Comment by yakagami** [comment]
@Uvais-Mohammad  Thanks. I will take a look soon. Does this code run on it's own or are there missing code? It will be easier if I can get a full minimal app starting with main() so I can test it.

**Comment by Uvais-Mohammad** [comment]
@yakagami Here it is.

```
import 'package:flutter/material.dart';
import 'package:zoom_view/zoom_view.dart';

void main() async {
  runApp(const MaterialApp(
    home: SampleItemListView(),
  ));
}

class SampleItemListView extends StatefulWidget {
  const SampleItemListView({
    super.key,
  });

  @override
  State<SampleItemListView> createState() => _SampleItemListViewState();
}

class _SampleItemListViewState extends State<SampleItemListView> {
  final controller = ScrollController();

  final items = List.generate(100, (index) => 'item $index');

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Sample Items'),
      ),
      body: ColoredBox(
        color: Colors.white,
        child: ZoomListView(
          child: ListView.builder(
            controller: controller,
            restorationId: 'sampleItemListView',
            itemCount: items.length,
            itemBuilder: (BuildContext context, int index) {
              final item = items[index];

              return Center(
                child: ListTile(
                    title: Text('SampleItem $item',
                        style: const TextStyle(color: Colors.black)),
                    leading: const CircleAvatar(
                      foregroundImage:
                          AssetImage('assets/images/flutter_logo.png'),
                    ),
                    onTap: () {}),
              );
            },
          ),
        ),
      ),
    );
  }
}

```

**Comment by yakagami** [comment]
Hi @Uvais-Mohammad I have time to look into this now. Seems like the ZoomWidget needs to be in a column with expanded in order to work. 

```dart

import 'package:flutter/material.dart';
import 'package:zoom_view/zoom_view.dart';

void main() async {
  runApp(const MaterialApp(
    home: SampleItemListView(),
  ));
}

class SampleItemListView extends StatefulWidget {
  const SampleItemListView({
    super.key,
  });

  @override
  State<SampleItemListView> createState() => _SampleItemListViewState();
}

class _SampleItemListViewState extends State<SampleItemListView> {
  final controller = ScrollController();

  final items = List.generate(100, (index) => 'item $index');

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Sample Items'),
      ),
      body: ColoredBox(
        color: Colors.white,
        child: Column(
          children: [
            Expanded(
              child: ZoomListView(
                child: ListView.builder(
                  controller: controller,
                  restorationId: 'sampleItemListView',
                  itemCount: items.length,
                  itemBuilder: (BuildContext context, int index) {
                    final item = items[index];

                    return Center(
                      child: ListTile(
                          title: Text('SampleItem $item',
                              style: const TextStyle(color: Colors.black)),
                          leading: const CircleAvatar(
                            foregroundImage:
                            AssetImage('assets/images/flutter_logo.png'),
                          ),
                          onTap: () {}),
                    );
                  },
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

```

Not sure if you are still using this package but thanks for the report and I will try to look into this issue more to see if there is a better solution where the developer doesn't need to add Column and Expanded themselves.

**Comment by yakagami** [comment]
should be fixed in 0.0.13
